driver-pulsar/deploy/provision-pulsar-aws.tf

```diff
@@ -126,33 +126,35 @@ resource "aws_instance" "zookeeper" {
   vpc_security_group_ids = ["${aws_security_group.benchmark_security_group.id}"]
   count                  = "${var.num_instances["zookeeper"]}"
 
-  tags {
+  tags = {
     Name = "zk-${count.index}"
   }
 }
 
-resource "aws_instance" "pulsar" {
+resource "aws_spot_instance_request" "pulsar" {
   ami                    = "${var.ami}"
   instance_type          = "${var.instance_types["pulsar"]}"
   key_name               = "${aws_key_pair.auth.id}"
   subnet_id              = "${aws_subnet.benchmark_subnet.id}"
   vpc_security_group_ids = ["${aws_security_group.benchmark_security_group.id}"]
   count                  = "${var.num_instances["pulsar"]}"
+  wait_for_fulfillment   = true
 
-  tags {
+  tags = {
     Name = "pulsar-${count.index}"
   }
 }
 
-resource "aws_instance" "client" {
+resource "aws_spot_instance_request" "client" {
```

Comment:
can we revert this part about "spot_instances" ? it can be a follow up work (or we can just keep it in our repo)